src/psyclone/tests/psyad/tl2ad_test.py

```diff
@@ -69,23 +77,53 @@ def test_generate_adjoint_str(caplog):
         "end program test\n")
 
     with caplog.at_level(logging.INFO):
-        result = generate_adjoint_str(tl_code)
+        result, test_harness = generate_adjoint_str(tl_code)
 
     assert caplog.text == ""
     assert expected in result
+    assert test_harness is None
 
     with caplog.at_level(logging.DEBUG):
-        result = generate_adjoint_str(tl_code)
+        result, test_harness = generate_adjoint_str(tl_code)
 
     assert "DEBUG    psyclone.psyad.tl2ad:tl2ad.py:58" in caplog.text
     assert tl_code in caplog.text
     assert "DEBUG    psyclone.psyad.tl2ad:tl2ad.py:74" in caplog.text
     assert expected in caplog.text
     assert expected in result
+    assert test_harness is None
+
+
```

Comment:
Sorry to be a pain but could you
a) Add a comment before each set of tests that is associated with a particular class/method/function (as is already done for `generate_adjoint_str`) e.g. you missed out `_find_container`
b) Order the tests so that the tests associated with a particular class/method/function are within the comment region, as they are not currently all in order. For example `test_generate_adjoint_str_generate_harness` should be in the first section.

Reply:
Done.

Reply:
Thanks.